### ISSUE TITLE
`super()` with other methods and properties

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#908]
# Rule request

## Thesis

An interesting idea I got this evening. I usually see code like this:

```python
class Child(Parent):
    def some(self):
        original = super().some()
        ...
```

And it is perfectly fine. That's why we have `super()` in the first place. But, I also saw this code today:

```python
class Child(Parent):
    def some(self):
        other = super().other()
        ...
```

Wait, what? Why `Child.some()` uses `super().other()` and not `self.other()`?
I think that it is important to enforce the following rule: one can only use `super()` method that matches the following context. Like: `super().some()` and `super().some` in `Child.some()`, and `super().prop` and `super().prop()` in `Child.prop` (if we are dealing with properties).

## Reasoning

It enforces better OOP design. And better responsibility distribution.

**Comment by cmppoon** [comment]
I want to help with this one. Can I take this?

**Comment by sobolevn** [comment]
Sure, thanks a lot! @cmppoon please, feel free to ask any questions you have.

**Comment by sobolevn** [comment]
Closed by #922
